framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -1429,26 +1430,28 @@ MeshDiagnosticsGenerator::checkNonMatchingEdges(const std::unique_ptr<MeshBase>
     bounding_box_map.insert({elem, boundingBox});
   }
 
+  std::unique_ptr<PointLocatorBase> point_locator = PointLocatorBase::build(TREE_ELEMENTS, *mesh);
```

Comment:
Would `point_locator = mesh->sub_point_locator()` not work here?  In this use case I admit it's unlikely either that someone else has already constructed a point locator we can use or that someone else will take the opportunity to reuse the one we construct, but in general it's better to leave the point locator in the hands of the `Mesh` unless we need some particular non-default setting.